hazelcast/src/main/java/com/hazelcast/config/Config.java

```diff
@@ -1328,30 +2553,59 @@ public HotRestartPersistenceConfig getHotRestartPersistenceConfig() {
     }
 
     /**
-     * Sets hot restart configuration.
+     * Sets the Hot Restart configuration.
      *
-     * @param hrConfig hot restart configuration
-     * @return Config
+     * @param hrConfig Hot Restart configuration
+     * @return this config instance
+     * @throws NullPointerException if the {@code hrConfig} parameter is {@code null}
      */
     public Config setHotRestartPersistenceConfig(HotRestartPersistenceConfig hrConfig) {
         checkNotNull(hrConfig, "Hot restart config cannot be null!");
         this.hotRestartPersistenceConfig = hrConfig;
         return this;
     }
 
+    /**
+     * Returns the external managed context. This context is used to
+     * initialize user supplied objects.
+     *
+     * @return the managed context
+     */
     public ManagedContext getManagedContext() {
         return managedContext;
     }
 
+    /**
+     * Sets the external managed context. This context is used to
+     * initialize user supplied objects.
+     *
+     * @param managedContext the managed context
+     * @return this config instance
+     */
     public Config setManagedContext(final ManagedContext managedContext) {
         this.managedContext = managedContext;
         return this;
     }
 
+    /**
+     * Returns the user supplied context. This context can then be used to
+     * obtain some object instances from objects that have
```

Comment:
Didn't you mean "same"?

Reply:
`Some`, as in "if you have a reference to the HazelcastInstance, you can get to these objects". Users usually put some external resources here like JDBC connections.

Reply:
Then maybe 
> Returns the managed context, in which this config was used to create {@link HazelcastInstance}. It can be used to obtain other managed beans from the same context.

Reply:
Sorry, this sounds very confusing. For one, this context isn't managed in any way. It's just a bucket for some objects that the user might get to through the hazelcast instance. Simple as that. It's hard to say we are "using" it to create the instance though - the instance only keeps a reference to it. It does not expect any object to be in there.